bat-utils/lib/runtime-wallet.js

```diff
@@ -161,39 +163,49 @@ Wallet.prototype._redeem = async function (info, txn, signature) {
 
   payload = {
     grants: [],
-    wallet: underscore.pick(info, [ 'provider', 'providerId' ]),
-    txn: txn,
-    signature: signature
+    // TODO might need paymentId later
+    wallet: underscore.pick(info, [ 'altcurrency', 'provider', 'providerId' ]),
+    transaction: Buffer.from(JSON.stringify(underscore.pick(signature, [ 'headers', 'octets' ]))).toString('base64')
   }
   grantIds = []
-  for (let entry of entries) {
-    entry.probi = entry.probi.toString()
-    payload.grants.push(underscore.pick(entry, [ 'altcurrency', 'probi', 'grantId', 'promotionId', 'grantSignature' ]))
-    grantIds.push(entry.grantId)
-
-    balance = balance.plus(new BigNumber(entry.probi.toString()))
+  let grantTotal = new BigNumber(0)
+  for (let grant of grants) {
+    payload.grants.push(grant.token)
+    grantIds.push(grant.grantId)
+
+    const grantContent = braveUtils.extractJws(grant.token)
+    const probi = new BigNumber(grantContent.probi)
+    balance = balance.plus(probi)
+    grantTotal = grantTotal.plus(probi)
     if (balance.greaterThanOrEqualTo(desired)) break
   }
 
-  result = await braveHapi.wreck.post(this.runtime.config.redeemer.url + '/v1/grants', {
-    headers: {
-      authorization: 'Bearer ' + this.runtime.config.redeemer.access_token,
-      'content-type': 'application/json'
-    },
-    payload: JSON.stringify(payload),
-    useProxyP: true
-  })
-  if (Buffer.isBuffer(result)) try { result = JSON.parse(result) } catch (ex) { result = result.toString() }
-
-  state = {
-    $currentDate: { timestamp: { $type: 'timestamp' } },
-    $set: { redeemed: result }
+  if (this.runtime.config.redeemer.cardId) { // temporary local redemption for integration with browser
```

Comment:
at some point we will want to remove this, i suspect.

Reply:
yes indeed